packages/vm/core/evm/evmimpl/internal.go

```
@@ -83,6 +84,17 @@ func (bctx *blockContext) mintBlock() {
 	bctx.emu.MintBlock()
 }
 
+func getTracer(ctx isc.Sandbox, bctx *blockContext) tracers.Tracer {
+	tracer := ctx.EVMTracer()
+	if tracer == nil {
+		return nil
+	}
+	if int(tracer.TxIndex) != len(bctx.txs) {
+		return nil
```

Should we maybe return an error in that case instead of no tracer?